docs/content/developer/workshops/gas-station-workshop.mdx

```diff
@@ -0,0 +1,375 @@
+
+    // 6. Send to Gas Station for co-sign + execution
+    const txBytesBase64 = btoa(String.fromCharCode(...new Uint8Array(unsignedTxBytes)));
+    const executeRes = await axios.post(
+      `${GAS_STATION_URL}/v1/execute_tx`,
+      { reservation_id, tx_bytes: txBytesBase64, user_sig: signature },
+      { headers: { Authorization: `Bearer ${GAS_STATION_AUTH}` } }
+    );
+
+    // 7. Report success to wallet
+    reportTransactionEffects(executeRes.data.effects);
+
+    // 8. Update UI
+    setPosts(prev => [{
+      content,
+      author: address!.slice(0, 10) + '...',
+      txid: executeRes.data.effects.transactionDigest,
+    }, ...prev]);
+    setContent('');
+    alert('Spark posted on-chain!');
+
+  } catch (err: any) {
+    console.error(err);
+    alert('Failed: ' + (err.response?.data?.error || err.message));
+  }
+};
+```
+
+### Explanation
+
+This is the complete sponsored transaction flow:
+
+- User builds intent
+- Reserves gas from sponsor
+- Attaches sponsor’s gas objects
+- Signs with wallet (Firefly popup)
+- Gas Station co-signs and submits
+- Success → on-chain forever!
+
+## 4.3 Final UI with Textarea & Feed
+
+Replace the return block with this beautiful final version (includes textarea, button, and feed).
+
+```tsx
+return (
+  <div style={{ maxWidth: 600, margin: '40px auto', fontFamily: 'system-ui' }}>
+    <h1>Spark Mini</h1>
+    <p>Gas-free microblogging on IOTA testnet</p>
+
+    {!address ? (
+      <div style={{ textAlign: 'center', marginTop: 80 }}>
+        <h2>Connect your wallet to post</h2>
+        <ConnectButton />
+      </div>
+    ) : (
+      <>
+        <p>Connected: {address.slice(0, 12)}...{address.slice(-8)}</p>
+        <textarea
+          value={content}
+          onChange={e => setContent(e.target.value)}
+          placeholder="What's your spark? (280 chars)"
+          maxLength={280}
+          rows={4}
+          style={{ width: '100%', padding: 12, fontSize: 16, borderRadius: 8, border: '1px solid #ccc' }}
+        />
+        <button
+          onClick={handlePost}
+          style={{ marginTop: 12, padding: '12px 24px', background: '#0068FF', color: 'white', border: 'none', borderRadius: 8, fontSize: 16 }}
+        >
+          Post Spark
+        </button>
+
+        <hr style={{ margin: '40px 0' }} />
+
+        {posts.map((p, i) => (
+          <div key={i} style={{ padding: 16, border: '1px solid #eee', borderRadius: 12, marginBottom: 12 }}>
+            <strong>{p.author}:</strong> {p.content}
+            <br />
+            <small>
+              <a href={`https://explorer.iota.org/testnet/transaction/${p.txid}`} target="_blank">
+                View on Explorer
+              </a>
+            </small>
+          </div>
+        ))}
+      </>
+    )}
+  </div>
+);
+```
+
+### Explanation
+
+Clean, responsive UI with live feed and explorer links. Every post is permanently stored on IOTA testnet.
+
+## Final Test
+
+- `npm run dev`
+- Connect Firefly
```

Comment:
No no no

Reply:
Yes, sorry, it appeared I was supposed to update it. @Thoralf-M